src/hotspot/share/utilities/bitMap.hpp

```diff
@@ -315,6 +322,15 @@ class BitMap {
   // aligned to bitsizeof(bm_word_t).
   idx_t get_next_one_offset_aligned_right(idx_t l_index, idx_t r_index) const;
 
+  static constexpr idx_t NotFound = ~(idx_t)0;
+  // Find the index of the first bit set scanning from r_index (inclusive) to
+  // l_index (inclusive) backwards.
```

Comment:
I don't think r_index should be inclusive.  I think that's surprising (such
ranges are usually half-open, and always so elsewhere in this class).  (It
caught me by surprise; I failed to notice that in the doc and was quite
confused when reviewing the implementation until I realized what was going
on.)  I also think it makes these operations harder to use. Having a closed
range forces the caller to deal with the unrepresentable empty range
explicitly.  A common use-case for such searches is to iterate over the bits;
the inclusive r_index makes that more awkward for the caller.  Obviously this
affects everything else in this change, so I haven't looked at the code very
much yet.

Reply:
I see your point, and I'm not against that change.

In ZGC code the equivalent method also seems `r_index`-inclusive. @stefank , any opinions?